config/parse.go

```diff
@@ -311,15 +310,15 @@ func resolveHandlers(c *Config) error {
 			logger = logger.WithField("transformer", t)
 
 			var nextT skogul.Transformer
-			if c.Transformers[t] != nil {
+			if c.Transformers[t.Name] != nil {
 				logger.Debug("Using predefined transformer")
-				nextT = c.Transformers[t].Transformer
-			} else if t == "templater" {
+				nextT = c.Transformers[t.Name].Transformer
+			} else if t.Name == "templater" {
 				logger.Debug("Using templating transformer")
 				nextT = transformer.Templater{}
 			} else {
 				logger.Error("Unknown transformer")
-				return skogul.Error{Source: "config", Reason: fmt.Sprintf("Unknown transformer %s", t)}
+				return skogul.Error{Source: "config", Reason: fmt.Sprintf("Unknown transformer %s", t.Name)}
 			}
 			h.Handler.Transformers = append(h.Handler.Transformers, nextT)
```

Comment:
Hmm, interesting.... Ok, let's leave the hmm-interesting out of it for now:

There's no point looking up c.Transformers[t.Name] since we have t.T already. So this entire if-thing from 311 to 323 can be shortened down to:

h.Handler.Transformers = append(h.Handler.Transformers, t.T)

This does introduce a change in feature, since we're dropping the "templater" default. But I think that's the correct thing to do, and if we want implicit objects like that, we should instead formalize it. And I suspect it's already broken because resolveTransformers wont be able to resolve the templater transformer anyway.

Reply:
To make sure I've understood you correctly: It's fine to drop support for the templating transformer? And if it's to be used, it should be reimplemented to work more in line with how other transformers work?

Reply:
Yeah, you can still specify it in "transformers" today, just "type": "templating" with no other info. So it's not dropping support, it's just bringing it in line with the others, now that there are a whole bunch of them.